### PR TITLE
[#3] abstract GSetGenerator

### LINKED ISSUE #3: Abstract out GSetGenerator for tests
Today in NNA all the unit tests utilize a class called "GSetGenerator".

This is the main class that is responsible for generating the INode tree (its a Set really, in the case of NameNode). Once the Set is generated its passed into NNA via a special loader method and the unit tests carry on.

It would be nice if this GSetGenerator was abstracted out as an interface and we just had each Hadoop version provide the implementation; similar to how the VersionContext and VersionInterface work today.

### PR DESCRIPTION
**Pull request by jhsenjaliya**
Make sure you have checked all steps below.

### GitHub Issue
Fixes #3 

### Checklist:

- [ ] This pull request updates the documentation
- [x] This pull request changes the code
- [x] Title of the PR is of format (example) : `[#25] Add Pull Request Template`, where `[#25] is the GitHub issue number`

### What is the purpose of this pull request?
provide abstraction for GSetGenerator class so that every hadoop version dont have to implement it from scratch.

### How was this change validated?
by running all the tests

### Commit Guidelines
- [x] My commits all reference GH issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: src/test/java/org/apache/hadoop/hdfs/server/namenode/GSetGeneratorBase.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package org.apache.hadoop.hdfs.server.namenode;
```
[the remark this answers is not in the document]
Comment (r39132) [reply src/test/java/org/apache/hadoop/hdfs/server/namenode/GSetGeneratorBase.java]: Is this an older comment that GH somehow didn't address?

[the remark this answers is not in the document]
Comment (zero45) [reply src/test/java/org/apache/hadoop/hdfs/server/namenode/GSetGeneratorBase.java]: Yeah the address is there now in the branch.

**Comment by zero45** [comment]
@jhsenjaliya Can you rebase your changes? Would be nice to get Travis build kicked off too.

**Comment by zero45** [comment]
@jhsenjaliya  -- Seems you need to run `spotlessJavaApply` on some builds and on others its failing because StoragePolicies do not exist in those versions and yet you put it in the BASE. This is why `VersionContext.java` has been split across versions.

**Comment by jhsenjaliya** [comment]
ya looking at this now...trying to commonalize (just found new english word :) ) as much code as possible into base class, looks like the policyIds are the only class that needs to be replicated to version specific classes.

File: src/test/2.6.1/org/apache/hadoop/hdfs/server/namenode/GSetGenerator.java [hunk 2]
```diff
@@ -19,40 +19,21 @@
 
 package org.apache.hadoop.hdfs.server.namenode;
 
-import static org.apache.hadoop.hdfs.protocol.HdfsConstants.ALLSSD_STORAGE_POLICY_ID;
-import static org.apache.hadoop.hdfs.protocol.HdfsConstants.COLD_STORAGE_POLICY_ID;
-import static org.apache.hadoop.hdfs.protocol.HdfsConstants.HOT_STORAGE_POLICY_ID;
-import static org.apache.hadoop.hdfs.protocol.HdfsConstants.MEMORY_STORAGE_POLICY_ID;
-import static org.apache.hadoop.hdfs.protocol.HdfsConstants.ONESSD_STORAGE_POLICY_ID;
-import static org.apache.hadoop.hdfs.protocol.HdfsConstants.WARM_STORAGE_POLICY_ID;
+import static org.apache.hadoop.hdfs.protocol.HdfsConstants.*;
```
Comment (zero45) [inline src/test/2.6.1/org/apache/hadoop/hdfs/server/namenode/GSetGenerator.java]: Oh man; google-java-format allows this? :(

Comment (jhsenjaliya) [reply src/test/2.6.1/org/apache/hadoop/hdfs/server/namenode/GSetGenerator.java]: offcourse, its all human that creates complexity, google-java-format likes anything that is logical..... :)

**Comment by zero45** [comment]
Thank you @jhsenjaliya!

**Pull request merged**